metals/src/main/scala/scala/meta/internal/implementation/SuperMethodProvider.scala

```diff
@@ -1,295 +1,27 @@
 package scala.meta.internal.implementation
 
-import scala.collection.{mutable => m}
-
-import scala.meta.internal.metals.codelenses.SuperMethodCodeLens.LensGoSuperCache
-import scala.meta.internal.semanticdb.ClassSignature
-import scala.meta.internal.semanticdb.MethodSignature
-import scala.meta.internal.semanticdb.Scala._
 import scala.meta.internal.semanticdb.SymbolInformation
-import scala.meta.internal.semanticdb.SymbolOccurrence
-import scala.meta.internal.semanticdb.TextDocument
-import scala.meta.internal.semanticdb.TypeRef
-import scala.meta.internal.semanticdb.TypeSignature
 
 object SuperMethodProvider {
 
   def findSuperForMethodOrField(
-      methodSymbolInformation: SymbolInformation,
-      documentWithPath: TextDocumentWithPath,
-      symbolRole: SymbolOccurrence.Role,
-      findSymbol: String => Option[SymbolInformation],
-      cache: LensGoSuperCache
+      methodSymbolInformation: SymbolInformation
   ): Option[String] = {
-    for {
-      methodSignature <- isDefinitionOfMethodField(
-        symbolRole,
-        methodSymbolInformation
-      )
-      superSymbol <- findSuperForMethodOrFieldChecked(
-        methodSymbolInformation,
-        methodSignature,
-        documentWithPath,
-        cache,
-        findSymbol
-      )
-    } yield superSymbol
+    getSuperMethodHierarchy(
+      methodSymbolInformation
+    ).headOption
```

Comment:
A nitpick: it seems that it should be enough only to take a headOption and check that it doesn't belong to stopSymbols.
Smth like: `methodSymbolInformation.overriddenSymbols.headOption.filter(isNonStopSymbol)`

Reply:
Actually, this check is done inside of `getSuperMethodHierarchy` :)

Reply:
Yep, I've seen that.
The only difference that it will be just slightly effective as there is no need to filter a full hierarchy chain in this case

Reply:
Ah, I got it!
addressed in https://github.com/scalameta/metals/pull/3045/commits/0dbbabac20bd796c823abcdca597647299a4540e